main/src/main/java/de/blinkt/openvpn/core/OpenVPNService.java

```diff
@@ -1033,7 +1054,16 @@ private void setAllowedVpnPackages(Builder builder) {
             }
         }
 
-        for (String pkg : mProfile.mAllowedAppsVpn) {
+        HashSet<String> mAllowedAppsVpn;
+
+        if ( getConnectivityByMobile(this) && mProfile.mAllowAppMobile) {
+            mAllowedAppsVpn = mProfile.mAllowedAppsVpnMobile;
+            VpnStatus.logInfo("Mobile data is active.");
```

Comment:
That log message in itself is not very useful. I am sure it was useful for debugging.